tauri-app/src-tauri/src/commands/fork.rs

```diff
@@ -0,0 +1,191 @@
+use alloy_dyn_abi::JsonAbiExt;
+use alloy_json_abi::Error;
+use forker::*;
+use once_cell::sync::Lazy;
+use reqwest::Client;
+use revm::primitives::Bytes;
+use std::{collections::HashMap, sync::Mutex};
+
+/// static hashmap of fork evm instances, used for caching instances between runs
+pub static FORKS: Lazy<Mutex<HashMap<String, ForkedEvm>>> =
+    Lazy::new(|| Mutex::new(HashMap::new()));
+
+/// hashmap of cached error selectors    
+pub static SELECTORS: Lazy<Mutex<HashMap<Vec<u8>, Error>>> =
+    Lazy::new(|| Mutex::new(HashMap::new()));
+
+#[tauri::command]
+pub async fn fork_call(
+    fork_url: &str,
+    fork_block_number: u64,
+    from_address: &[u8],
+    to_address: &[u8],
+    calldata: &[u8],
+) -> Result<Result<Bytes, String>, String> {
+    let result = {
+        // lock static FORKS
+        let mut forks = FORKS.lock().unwrap();
+
+        // build key from fork url and block number
+        let key = fork_url.to_owned() + &fork_block_number.to_string();
+
+        // fork from the provided url, if it is cached, use it, if not create it, and cache it in FORKS
+        let forked_evm = if let Some(v) = forks.get_mut(&key) {
```

Comment:
i think we can use entry here to handle the default empty case https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html